### PR TITLE
Fix more missing #includes for GCC 11

### PR DESCRIPTION
**Pull request by Conan-Kudo**
This is from an effort in Fedora to prepare for the GCC 11 rebase for Fedora 34.

Followup to #1771.

### CODE REVIEW

File: src/server/input/default_input_manager.cpp [hunk 1]
```diff
@@ -16,6 +16,8 @@
  * Authored by: Andreas Pokorny <andreas.pokorny@canonical.com>
  */
 
+#include <memory>
+
 #include "default_input_manager.h"
```
Comment (AlanGriffiths) [inline src/server/input/default_input_manager.cpp:21]: We have a policy of including the "corresponding" header first (in this case `#include "default_input_manager.h"` should be the first include).

Please put `#include <memory>` after `#include <future>`

File: tests/unit-tests/input/test_default_input_manager.cpp [hunk 2]
```diff
@@ -16,6 +16,7 @@
  * Authored by: Andreas Pokorny <andreas.pokorny@canonical.com>
  */
 
+#include <memory>
 #include "src/server/input/default_input_manager.h"
```
Comment (AlanGriffiths) [inline tests/unit-tests/input/test_default_input_manager.cpp:20]: Please put `#include <memory>` after `#include <list>`, not before the project headers.

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
Thanks for this. Could you please re-order these includes to be more consistent with the project style.

**Comment by Conan-Kudo** [comment]
@AlanGriffiths I believe I've addressed your feedback now.

**Review by AlanGriffiths — APPROVED** [review]
LGTM. Thanks!

bors r+

**Pull request merged**